src/views/Predictions/components/RoundCard/SetPositionCard.tsx

```diff
@@ -86,30 +72,46 @@ const SetPositionCard: React.FC<SetPositionCardProps> = ({ position, togglePosit
   const { toastError } = useToast()
   const predictionsContract = usePredictionsContract()
 
-  const balanceDisplay = getBnbAmount(bnbBalance).toString()
-  const maxBalance = getBnbAmount(bnbBalance.gt(dust) ? bnbBalance.minus(dust) : bnbBalance).toNumber()
+  const balanceDisplay = useMemo(() => {
+    return getBnbAmount(bnbBalance).toString()
+  }, [bnbBalance])
+  const maxBalance = useMemo(() => {
+    return getBnbAmount(bnbBalance.gt(dust) ? bnbBalance.minus(dust) : bnbBalance)
+  }, [bnbBalance])
+  const minBetAmountBalance = useMemo(() => {
+    return getBnbAmount(minBetAmount)
+  }, [minBetAmount])
+
   const valueAsBn = new BigNumber(value)
 
-  const percentageOfMaxBalance = valueAsBn.div(maxBalance).times(100).toNumber()
-  const percentageDisplay = getPercentDisplay(percentageOfMaxBalance)
   const showFieldWarning = account && valueAsBn.gt(0) && errorMessage !== null
-  const minBetAmountBalance = getBnbAmount(minBetAmount).toNumber()
 
-  const handleChange = (input) => {
-    setValue(input)
-  }
+  const [percent, setPercent] = useState(0)
 
-  const handleSliderChange = (newValue: number) => {
-    setValue(newValue.toString())
+  const handleInputChange = (input: string) => {
+    if (input) {
+      const percentage = Math.floor(new BigNumber(input).dividedBy(maxBalance).multipliedBy(100).toNumber())
+      setPercent(Math.min(percentage, 100))
```

Comment:
Why `Math.floor` why `Math.min`?

Reply:
Min is for when user enters value that exceeds max balance, floor is for not to show too many decimals in percentage